packages/hint-no-broken-links/src/hint.ts

```diff
@@ -169,15 +146,56 @@ export default class NoBrokenLinksHint implements IHint {
             fetchedUrls.push({ statusCode: fetchEnd.response.statusCode, url: fetchEnd.resource });
         };
 
-        context.on('element::img', validateElementSrcs);
-        context.on('element::a', validateElementSrcs);
-        context.on('element::audio', validateElementSrcs);
-        context.on('element::video', validateElementSrcs);
-        context.on('element::link', validateElementSrcs);
-        context.on('element::script', validateElementSrcs);
-        context.on('element::source', validateElementSrcs);
-        context.on('element::track', validateElementSrcs);
-        context.on('element::object', validateElementSrcs);
+        const validateCollectedUrls = async (event: TraverseEnd) => {
+            const { resource } = event;
+            const pageDOM: IAsyncHTMLDocument = context.pageDOM as IAsyncHTMLDocument;
+            const baseTags: Array<IAsyncHTMLElement> = await pageDOM.querySelectorAll('base');
+            const hrefAttribute = (baseTags.length === 0) ? null : baseTags[0].getAttribute('href');
+
+            const reports: Array<Promise<void>> = collectedElementsWithUrls.reduce<Promise<void>[]>((accumulatedReports, [element, urls]) => {
+                return [...accumulatedReports, ...urls.map(async (url) => {
```

Comment:
Is there any particular reason of using spreading instead of `concat`? Stylistic is a good answer btw.

Reply:
IMO `urls.map` should be move to its own variable/method and then `return [...accumulatedReports, ...reports]` or even `return [...accumulatedReport, ...getReports(urls)]`.
I think it would be more readable.

Reply:
That was just a stylistic decision. Let me know if you want me to change it to use `concat()` instead.

Reply:
Maybe just do what @sarvaje proposes of extracting `urls.map` to its own variable to make it a bit clearer 😊